sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -357,7 +357,7 @@ def _fit(self, X, y, incremental=False):
         activations = [X]
         activations.extend(np.empty((batch_size, n_fan_out))
                            for n_fan_out in layer_units[1:])
-        deltas = [np.empty_like(a_layer) for a_layer in activations]
+        deltas = [np.empty_like(a_layer) for a_layer in activations[1:]]
```

Comment:
This can be:

```suggestion
        deltas = [None] * (len(activations) - 1)
```

because of how we do backprop only requires a list with `len(activations) - 1` items.

https://github.com/scikit-learn/scikit-learn/blob/4becb71a0469065bfe1c722f52e753d01a661f37/sklearn/neural_network/multilayer_perceptron.py#L240